services/backend/private-cloud.ts

```diff
@@ -43,6 +44,41 @@ export async function deletePrivateCloudComment(
   return response.data;
 }
 
+export interface PrivateCloudProductAllCriteria {
+  search: string;
+  page: number;
+  pageSize: number;
+  ministry: string;
+  cluster: string;
+  includeInactive: boolean;
+  sortKey: string;
+  sortOrder: string;
+}
+
+export interface PrivateCloudProductSearchCriteria extends PrivateCloudProductAllCriteria {
+  page: number;
+  pageSize: number;
+}
+
+export async function searchPriviateCloudProducts(data: PrivateCloudProductSearchCriteria) {
+  const result = await instance.post(`private-cloud/products/search`, data).then((res) => {
+    return res.data;
+  });
+
+  return result as PrivateCloudProductSearchPayload;
+}
+
+export async function downloadPriviateCloudProducts(data: PrivateCloudProductAllCriteria) {
+  const result = await instance.post(`private-cloud/products/download`, data, { responseType: 'blob' }).then((res) => {
+    if (res.status === 204) return false;
+
+    downloadFile(res.data, 'product-data.csv');
```

Comment:
Would it be possible to name the downloaded file with something like we currently have `private-cloud-products.csv` ? 
![Screenshot 2024-04-24 at 1 14 17 PM](https://github.com/bcgov/platform-services-registry/assets/145396323/0b59dbed-c221-48b7-b21f-deb10611adf7)

Reply:
@Amritpal-Nijjar yes, updated!